webapp/channels/src/widgets/button/button.tsx

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React, {forwardRef, memo, useMemo} from 'react';
+import type {ButtonHTMLAttributes, ReactNode} from 'react';
+
+import './button.scss';
+
+export type ButtonSize = 'xs' | 'sm' | 'md' | 'lg';
+export type ButtonEmphasis = 'primary' | 'secondary' | 'tertiary' | 'quaternary' | 'link';
+
+// Removed ButtonStyle type - now using inverted boolean prop
+
+export interface ButtonProps {
+
+    /** Button content */
+    children?: ReactNode;
+
+    /** Size variant of the button */
+    size?: ButtonSize;
+
+    /** Emphasis level of the button */
+    emphasis?: ButtonEmphasis;
+
+    /** Visual style variant - inverted for dark backgrounds */
+    inverted?: boolean;
+
+    /** Whether the button represents a destructive action */
+    destructive?: boolean;
+
+    /** Loading state - shows spinner and disables interaction */
+    loading?: boolean;
+
+    /** Icon to display before the button text */
+    iconBefore?: ReactNode;
+
+    /** Icon to display after the button text */
+    iconAfter?: ReactNode;
+
+    /** Whether the button should take full width of its container */
+    fullWidth?: boolean;
+
+    /** Fixed width for the button (e.g., "200px", "10rem") */
+    width?: string;
+}
+
+type ButtonHTMLProps = Omit<ButtonHTMLAttributes<HTMLButtonElement>, keyof ButtonProps>;
+
+export interface ButtonComponent extends React.ForwardRefExoticComponent<ButtonProps & ButtonHTMLProps & React.RefAttributes<HTMLButtonElement>> {
```

Comment:
I'm not sure about these types, but I've spent enough time pointlessly fighting with TS types to know that it's not worth looking at now. I'll leave this as a reminder to maybe look at later 😛